src/Core/src/Handlers/WebView/WebViewHandler.Android.cs

```diff
@@ -56,16 +68,23 @@ public static void MapWebViewSettings(WebViewHandler handler, IWebView webView)
 
 		public static void MapGoBack(WebViewHandler handler, IWebView webView, object? arg)
 		{
+			if (handler.NativeView.CanGoBack())
+				handler.CurrentWebNavigationEvent = WebNavigationEvent.Back;
```

Comment:
The event state appears to get stuck with whatever it was set to.

- go to gallery for web view
- navigate forward
- click back
- now click any links on the page

You'll get a navigating event with a CurrentWebNavigationEvent set to `Back`

Maybe this call can just be centralized into the handler?

![image](https://user-images.githubusercontent.com/5375137/153514318-b0b0434b-b0d4-4a8d-83a5-7c6fc9153764.png)

Reply:
Found the issue, this is the pending piece of code: https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Renderers/FormsWebViewClient.cs#L66